### DISCUSSION TITLE
Colors change after composite

### DISCUSSION THREAD

**Discussion by bceyssens** [ImageMagick/ImageMagick#4032] [Help] [answered]
I'm trying to composite 2 images:
https://user-images.githubusercontent.com/7702573/128078600-2a14c7a6-0fac-455b-a4cf-6d577a5cbe33.png
https://user-images.githubusercontent.com/7702573/128078611-bdfc3639-a1a7-44c5-a5bd-194a7641df9b.png

With the command `convert background.png subject.png -composite result.jpg` this results into:
https://user-images.githubusercontent.com/7702573/128078609-79a453c6-a338-4753-bd53-9fc5214f3061.jpg

The colors of the subject image have shifted (much lighter than original image). I think it has something to do with the colorspace, but I can't find a proper solution to this problem. Any help is welcome.

**Comment by snibgo** [comment, marked as answer]
The problem is that your first image is encoded as linear RGB, and the second image is encoded as non-linear sRGB. So a direct composite is somewhat meaningless because different pixels will use different encodings. Instead, they should be converted to the same colorspace before the composite.

As they are both defined by embedded ICC profile, we should use "-profile" to convert them. Windows syntax:
```
magick ^
  128078600-2a14c7a6-0fac-455b-a4cf-6d577a5cbe33.png ^
  128078611-bdfc3639-a1a7-44c5-a5bd-194a7641df9b.png ^
  -profile sRGB.icc ^
  -composite ^
  out.jpg
```
My command assumes sRGB.icc is in your current directory. If it isn't, provide the full path. The IM installation directory probably contains a copy of sRGB.icc.

**Reply by bceyssens** [reply]
Works perfectly, thanks!
